beacon-chain/rpc/eth/beacon/blinded_blocks.go

```diff
@@ -233,13 +234,46 @@ func (bs *Server) SubmitBlindedBlockSSZ(ctx context.Context, req *ethpbv2.SSZCon
 	if err != nil {
 		return &emptypb.Empty{}, status.Errorf(codes.Internal, "Could not create unmarshaler: %v", err)
 	}
-	block, err := unmarshaler.UnmarshalBlindedBeaconBlock(req.Data)
-	if err != nil {
-		return &emptypb.Empty{}, status.Errorf(codes.Internal, "Could not unmarshal request data into block: %v", err)
-	}
 
 	switch forkVer {
+	case bytesutil.ToBytes4(params.BeaconConfig().DenebForkVersion):
+		blkContent := &ethpbv2.SignedBlindedBeaconBlockContentsDeneb{}
+		if err := blkContent.UnmarshalSSZ(req.Data); err != nil {
+			return &emptypb.Empty{}, status.Errorf(codes.Internal, "Could not unmarshal ssz block contents: %v", err)
+		}
+		blindedBlock, err := migration.BlindedDenebToV1Alpha1SignedBlock(blkContent.SignedBlindedBlock)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "Submitted block is not blinded: %v", err)
```

Comment:
hmm I think this should be right , instead of checking the is blinded function it's trying to migrate the type but if it's empty or doesn't marshal correctly I think we can say this. what do you think.